### PR TITLE
Issue 19440 apps button

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**
This adds interpolation capabilities (achieved via velocity) to the apps yml files.

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/view/AppsInterpolationTest.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        final List<Map<String, String>> list = ImmutableList.of(
+                ImmutableMap.of("label", "-", "value", ""),
+                ImmutableMap.of("label", "$uno", "value", "1"),
+                //This entry is required to be mutable since the selected attribute gets removed internally.
+                new HashMap<>(ImmutableMap.of("label", "dos", "value", "2", "selected", "true")),
+                ImmutableMap.of("label", "tres", "value", "3")
+        );
+
+        final AppDescriptorDataGen dataGen = new AppDescriptorDataGen()
+                .stringParam("param1", false, true, "lol", "name is `$param1.name`.", "This is `$param2.value`'s hint.")
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/view/AppsInterpolationTest.java:55]: Codacy found an issue: [The String literal "param1" appears 5 times in this file; the first occurrence is on line 55](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6392890)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/view/ViewUtil.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+                            velocityContext.put(name, new SecretHolder(name, label, hint, value));
+                        }
+                    }
+                }
+            }
+
+            final StringWriter stringWriter = new StringWriter();
+            template.merge(velocityContext, stringWriter);
+            return stringWriter.toString();
+        } catch (Exception e) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/view/ViewUtil.java:160]: Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6392890)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/view/ViewUtil.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.apache.velocity.Template;
+import org.apache.velocity.VelocityContext;
+import org.apache.velocity.runtime.RuntimeServices;
+import org.apache.velocity.runtime.RuntimeSingleton;
+import org.apache.velocity.runtime.parser.ParseException;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/view/ViewUtil.java:16]: Codacy found an issue: [Avoid unused imports such as 'org.apache.velocity.runtime.parser.ParseException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6392890)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/view/ViewUtil.java [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+    static String interpolateValues(final String inputJson) {
+        final StackContext context = ViewUtil.currentStackContext();
+        if (null == context) {
+            throw new IllegalStateException(
+                    "This method can only be called once a context has been previously created.");
+        }
+        try {
+            final RuntimeServices runtimeServices = RuntimeSingleton.getRuntimeServices();
+            final StringReader stringReader = new StringReader(inputJson);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/view/ViewUtil.java:119]: Codacy found an issue: [Ensure that resources like this StringReader object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6392890)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/view/ViewUtil.java [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.apache.velocity.Template;
+import org.apache.velocity.VelocityContext;
+import org.apache.velocity.runtime.RuntimeServices;
+import org.apache.velocity.runtime.RuntimeSingleton;
+import org.apache.velocity.runtime.parser.ParseException;
+import org.apache.velocity.runtime.parser.node.SimpleNode;
+
+class ViewUtil {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/view/ViewUtil.java:19]: Codacy found an issue: [All methods are static.  Consider using a utility class instead. Alternatively, you could add a private constructor or make the class abstract to silence this warning.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6392890)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/view/AppsInterpolationTest.java [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+     * Given scenario: we build an AppView json based on an AppDescriptor. This fires the serializers so that any embedded velocity code will br processed
+     * Expected Results: No velocity variable declared in the data model used to create the initial json such as `$myva` should make it in the result.
+     * Also we validate the replacements individually.
+     * @throws JsonProcessingException
+     */
+    @Test
+    public void Test_Interpolation() throws JsonProcessingException {
+
+        final List<Map<String, String>> list = ImmutableList.of(
+                ImmutableMap.of("label", "-", "value", ""),
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/view/AppsInterpolationTest.java:47]: Codacy found an issue: [The String literal "label" appears 6 times in this file; the first occurrence is on line 47](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6392890)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/view/AppsInterpolationTest.java [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+                ImmutableMap.of("label", "$uno", "value", "1"),
+                //This entry is required to be mutable since the selected attribute gets removed internally.
+                new HashMap<>(ImmutableMap.of("label", "dos", "value", "2", "selected", "true")),
+                ImmutableMap.of("label", "tres", "value", "3")
+        );
+
+        final AppDescriptorDataGen dataGen = new AppDescriptorDataGen()
+                .stringParam("param1", false, true, "lol", "name is `$param1.name`.", "This is `$param2.value`'s hint.")
+                .stringParam("param2", false, true, "none", "name is `$param2.name`.", "This is `$param1.value`'s hint.")
+                .selectParam("selectParam", true, list)
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/view/AppsInterpolationTest.java:57]: Codacy found an issue: [The String literal "selectParam" appears 5 times in this file; the first occurrence is on line 57](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6392890)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/view/AppsInterpolationTest.java [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+import java.util.List;
+import java.util.Map;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class AppsInterpolationTest {
+
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/view/AppsInterpolationTest.java:29]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6392890)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/view/AppsInterpolationTest.java [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+                ImmutableMap.of("label", "-", "value", ""),
+                ImmutableMap.of("label", "$uno", "value", "1"),
+                //This entry is required to be mutable since the selected attribute gets removed internally.
+                new HashMap<>(ImmutableMap.of("label", "dos", "value", "2", "selected", "true")),
+                ImmutableMap.of("label", "tres", "value", "3")
+        );
+
+        final AppDescriptorDataGen dataGen = new AppDescriptorDataGen()
+                .stringParam("param1", false, true, "lol", "name is `$param1.name`.", "This is `$param2.value`'s hint.")
+                .stringParam("param2", false, true, "none", "name is `$param2.name`.", "This is `$param1.value`'s hint.")
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/view/AppsInterpolationTest.java:56]: Codacy found an issue: [The String literal "param2" appears 5 times in this file; the first occurrence is on line 56](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6392890)

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/view/ViewUtil.java (hunk 5 above)
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/view/ViewUtil.java:19]: Some doc

**Review by jdotcms — APPROVED** [review]

**Pull request merged**